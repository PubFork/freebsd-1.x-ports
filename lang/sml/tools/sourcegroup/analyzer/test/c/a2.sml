(* datatypes *)

structure B = struct type b = bool end

structure A = struct
  datatype ('a, 'b) Map =
     Empty
   | Single of 'a * 'b
   | Flagged of 'a * 'b * B.b
  exception R
end

structure C = struct
  val x :('a,'b) A.Map = A.Empty

  exception E of B.b

  val y =
    case x of
       (A.Single (_,_)) => (A.Empty; print "foo\n")
     | _ => ()
end

structure D = struct
  exception F = C.E
end

structure E = struct
  exception X = A.R
end

(*
Seq
; Define s%B
; Define s%A
; = Ref s%B 
; Define s%C
; = Seq
; = ; Ref s%A s%A s%B s%A s%A 
; Define s%D
; = Ref s%C 
; Define s%E
; = Ref s%A 
*)
