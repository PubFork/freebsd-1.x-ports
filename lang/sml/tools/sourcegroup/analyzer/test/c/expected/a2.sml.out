source sml "c/a2.sml"
  export structure A B C D E ;

